### PR TITLE
Merge 2021-07-21

### PR DESCRIPTION
**Pull request by alexanderConstantinescu**
/assign @dcbw @trozet 

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/node/gateway_localnet.go [hunk 1]
```diff
@@ -64,24 +64,27 @@ func newLocalGateway(nodeName string, hostSubnets []*net.IPNet, gwNextHops []net
 		}
 		// END OCP HACK
 	} else {
-		bridgeName, uplinkName, macAddress, _, err := gatewayInitInternal(
-			nodeName, gwIntf, hostSubnets, gwNextHops, nodeAnnotator)
+		gwBridge, _, err := gatewayInitInternal(
```
Comment (trozet) [inline go-controller/pkg/node/gateway_localnet.go:67]: merge conflict resolution looks right to me

File: go-controller/pkg/node/gateway_shared_intf.go [hunk 2]
```diff
@@ -568,25 +590,32 @@ func newSharedGateway(nodeName string, subnets []*net.IPNet, gwNextHops []net.IP
 		return nil, fmt.Errorf("failed to setup MCS-blocking rules: %w", err)
 	}
 	// END OCP HACK
-
 	gw.readyFunc = func() (bool, error) {
```
Comment (trozet) [inline go-controller/pkg/node/gateway_shared_intf.go:593]: I don't think this is right, it should be:
```
	if exGwBridge != nil {
		gw.readyFunc = func() (bool, error) {
			ready, err := gatewayReady(gwBridge.patchPort)
			if err != nil {
				return false, err
			}
			exGWReady, err := gatewayReady(exGwBridge.patchPort)
			if err != nil {
				return false, err
			}
			return ready && exGWReady, nil
		}
	} else {
		gw.readyFunc = func() (bool, error) {
			return gatewayReady(gwBridge.patchPort)
		}
	}
```
https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/node/gateway_shared_intf.go#L580

@fedepaol can confirm

**Review by trozet — CHANGES_REQUESTED** [review]
/hold

File: go-controller/pkg/node/gateway_localnet.go [hunk 3]
```diff
@@ -64,24 +64,27 @@ func newLocalGateway(nodeName string, hostSubnets []*net.IPNet, gwNextHops []net
...
 			return nil, err
 		}
 
 		// the name of the patch port created by ovn-controller is of the form
 		// patch-<logical_port_name_of_localnet_port>-to-br-int
-		patchPort := "patch-" + bridgeName + "_" + nodeName + "-to-br-int"
+		patchPort := "patch-" + gwBridge.bridgeName + "_" + nodeName + "-to-br-int"
 
 		gw.readyFunc = func() (bool, error) {
 			return gatewayReady(patchPort)
```
Comment (trozet) [inline go-controller/pkg/node/gateway_localnet.go:78]: actually this isnt right:
https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/node/gateway_localnet.go#L69

**Comment by trozet** [comment]
@alexanderConstantinescu in the interest of time I started another one with what I think is the right merge conflict resolution:
https://github.com/openshift/ovn-kubernetes/pull/618

https://github.com/openshift/ovn-kubernetes/pull/618/files#diff-32663df3b1d1ffb0ecde35d8f44b6e4e2ae451b5712eaa054a1ee99fc07918fb

https://github.com/openshift/ovn-kubernetes/pull/618/files#diff-d3aa58d9b58a0a09264f072df46ab01d0501eb508c4656411ae2dc1ac68fb3c4R581

File: go-controller/pkg/node/gateway_shared_intf.go (hunk 2 above)
Comment (fedepaol) [reply go-controller/pkg/node/gateway_shared_intf.go:593]: Yes I confirm

**Pull request closed without merging**